packages/discovery-provider/src/tasks/entity_manager/entities/track.py

```diff
@@ -620,6 +621,38 @@ def delete_track(params: ManageEntityParameters):
     params.add_record(track_id, deleted_track)
 
 
+def download_track(params: ManageEntityParameters):
+    validate_track_tx(params)
+
+    track_id = params.entity_id
+    existing_track = params.existing_records["Track"][track_id]
```

Comment:
this isn't a `params.existing_record.get("Track")` situation?

Reply:
good call - adding a validator

Reply:
won't this throw a type error if `existing_records["Track"]` doesn't exist? i think that's what alec meant with the `.get`?

Reply:
yes, but in `validate_track_tx` we will throw an indexing validation error if that track doesn't exist

Reply:
ah sry i was going by commit, ignore me!